src/app/features/issue/providers/caldav/caldav-client.service.ts

```diff
@@ -0,0 +1,312 @@
+import {Injectable} from '@angular/core';
+import {CaldavCfg} from './caldav.model';
+// @ts-ignore
+import DavClient, {namespaces as NS} from 'cdav-library';
+// @ts-ignore
+import Calendar from 'cdav-library/models/calendar';
+// @ts-ignore
+import ICAL from 'ical.js';
+
+import {from, Observable, throwError} from 'rxjs';
+import {CaldavIssue} from './caldav-issue/caldav-issue.model';
+import {CALDAV_TYPE} from '../../issue.const';
+import {SearchResultItem} from '../../issue.model';
+import {SnackService} from '../../../../core/snack/snack.service';
+import {T} from '../../../../t.const';
+import {catchError} from 'rxjs/operators';
+import {HANDLED_ERROR_PROP_STR} from '../../../../app.constants';
+
+
+interface ClientCache {
+  client: DavClient;
+  calendars: Map<string, Calendar>;
+}
+
+@Injectable({
+              providedIn: 'root',
+            })
+export class CaldavClientService {
+
+  private _clientCache = new Map<string, ClientCache>();
+
+  constructor(
+    private readonly _snackService: SnackService
+  ) {
+  }
+
+  private _getXhrProvider(cfg: CaldavCfg) {
```

Comment:
We also have axios available as dependency, which might help to simplify things here.

Reply:
I got this xhr-provider-solution from the cdav-library repo (they've no documentation - but it was as a usage-example within an issue).  How would axios help here?